src/app/functions/server/onerep.ts

```diff
@@ -296,3 +300,27 @@ export async function getScanDetails(
   }
   return response.json() as Promise<Scan>;
 }
+
+export async function getAllScanResults(
+  profileId: number
+): Promise<ScanResult[]> {
+  const scanPagesAll = [];
+  const firstPage = await listScanResults(profileId, {
+    per_page: 100,
+  });
+  // Results are paginated, use per_page maximum and collect all pages into one result.
+  if (firstPage.meta.last_page > 1) {
+    let currentPage = 2;
+    while (currentPage <= firstPage.meta.last_page) {
+      const nextPage = await listScanResults(profileId, {
+        per_page: 100,
```

Comment:
Q: Does _this_ version need to pass `page: currentPage,` to track the current page number?

Reply:
Ugh yes thanks